flair/datasets/sequence_labeling.py

```diff
@@ -1425,6 +1430,233 @@ def __init__(
         )
 
 
+
+class CLEANCONLL(ColumnCorpus):
+    def __init__(
+            self,
+            base_path: Optional[Union[str, Path]] = None,
+            in_memory: bool = True,
+            **corpusargs,
+    ) -> None:
+        """
+        Initialize the CleanCoNLL corpus.
+
+        Args:
+            base_path: Base directory for the dataset. If None, defaults to flair.cache_root / "datasets".
+            in_memory: If True, keeps dataset in memory for faster training.
+        """
+        # Set the base path for the dataset
+        base_path = flair.cache_root / "datasets" if not base_path else Path(base_path)
+
+        # Define column format
+        columns = {0: "text",
+                   1: "pos",
+                   2: "nel",
+                   3: "ner*",
+                   4: "ner"}
+
+        # Define dataset name
+        dataset_name = self.__class__.__name__.lower()
+
+        # Define data folder path
+        data_folder = base_path / dataset_name
+
+        # Check if the train data file exists, otherwise download and prepare the dataset
+        train_set = data_folder / 'cleanconll.train'
+
+        if not train_set.exists():
+            print("CleanCoNLL files not found, so downloading and creating them.")
+
+            # Download and prepare the dataset
+            self.download_and_prepare_data(data_folder, base_path)
+
+        else:
+            print("Found files for CleanCoNLL in:", data_folder)
+
+        # Initialize the parent class with the specified parameters
+        super().__init__(
+            data_folder,
+            columns,
+            encoding="utf-8",
+            in_memory=in_memory,
+            document_separator_token="-DOCSTART-",
+            **corpusargs,
+        )
+
+    @staticmethod
+    def download_and_prepare_data(data_folder: Path):
+
+        def parse_patch(patch_file_path):
+            """
+            Parses a patch file and returns a structured representation of the changes.
+            """
+
+            changes = []
+            current_change = None
+
+            with open(patch_file_path, 'r') as patch_file:
```

Comment:
it's better to also specify an encoding when reading/writing files, as different os have different default encodings